src/ripple/protocol/impl/BuildInfo.cpp

```diff
@@ -38,20 +38,17 @@ char const* const versionString = "1.10.0-rc3"
 
 #if defined(DEBUG) || defined(SANITIZER) || defined(DEVELOP)
     "+"
+#define SEPARATOR ""
 #ifdef GIT_COMMIT_HASH
     GIT_COMMIT_HASH
-#if ! defined(DEVELOP)
-    "."
-#endif
+    #define SEPARATOR "."
```

Comment:
Won't this (and the rest of these) result in warnings about a redefinition of `SEPARATOR`?

Also, my _personal preference_ would be to have `DEBUG` and `DEVELOP` go _before_ the `git commit` hash.

Reply:
Warning? Yes. I offered this revision in a chat with @legleux.
```cpp
#define SEPARATOR "+"
#ifdef GIT_COMMIT_HASH
    SEPARATOR
    GIT_COMMIT_HASH
    #undef SEPARATOR
    #define SEPARATOR "."
#endif
#ifdef DEBUG
    SEPARATOR
    "DEBUG"
    #undef SEPARATOR
    #define SEPARATOR "."
#endif
#ifdef SANITIZER
    SEPARATOR
    BOOST_PP_STRINGIZE(SANITIZER)
    #undef SEPARATOR
    #define SEPARATOR "."
#endif
```